src/Oro/ORM/Query/AST/Functions/Cast.php

```diff
@@ -78,12 +70,16 @@ public function parse(Parser $parser)
     protected function checkType($type)
     {
         $type = strtolower(trim($type));
-        foreach ($this->supportedTypes as $supportedType) {
-            if (strpos($type, $supportedType) === 0) {
-                return true;
-            }
-        }
+        return Type::hasType($type);
+    }
 
-        return false;
+    /**
+     * Gets supported types
+     *
+     * @return array
+     */
+    protected function getSupportedTypes()
+    {
+        return array_keys(Type::getTypesMap());
```

Comment:
We can not use this types, as some of them are not supported by all DBs, that are supported by extensions